### PR TITLE
valgrind package: added lto variant and configured arguments for cray, mpi

### PR DESCRIPTION
**Pull request by francinelapid**

### CODE REVIEW

File: var/spack/repos/builtin/packages/valgrind/package.py [hunk 1]
```diff
@@ -47,13 +47,17 @@ class Valgrind(AutotoolsPackage, SourcewarePackage):
             description='Activates ubsan support for valgrind')
     variant('libs', default='shared,static', values=('shared', 'static'),
             multi=True, description='Build shared libs, static libs or both')
+    variant('lto', default=False,
+            description='Activates link-time optimization for valgrind')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/valgrind/package.py:51]:
```suggestion
    variant('lto', default=False, when='@3.15:',
            description='Activates link-time optimization for valgrind')
```

File: var/spack/repos/builtin/packages/valgrind/package.py [hunk 2]
```diff
@@ -47,13 +47,17 @@ class Valgrind(AutotoolsPackage, SourcewarePackage):
...
 
     conflicts('+ubsan', when='%apple-clang',
               msg="""
 Cannot build libubsan with clang on macOS.
 Otherwise with (Apple's) clang there is a linker error:
 clang: error: unknown argument: '-static-libubsan'
 """)
+    conflicts('+lto', when='@:3.14',
+              msg='Link-time Optimization only supported in 3.15.0 and above')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/valgrind/package.py:60]:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]
You can use a conditional variant instead of a variant + conflict

File: var/spack/repos/builtin/packages/valgrind/package.py [hunk 3]
```diff
@@ -82,6 +86,16 @@ def configure_args(self):
...
+        if spec.satisfies('+mpi'):
+            if self.spec.platform != 'cray':
+                options.append('--with-mpicc={0}'
+                               .format(self.spec['mpi'].mpicc))
+            else:
+                options.append("--with-mpicc=env['CC']")
+        if spec.satisfies('+lto'):
+            options.append('--enable-lto')
+        if spec.platform == 'cray':
+            env['SPACK_CC'] = "/opt/gcc/9.3.0/bin/gcc"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/valgrind/package.py:98]: We should also avoid hard-coded paths.

**Comment by alalazo** [comment]
@francinelapid Are you still working on this PR?

**Comment by francinelapid** [comment]
Yes, I'm still working on this PR.

**Comment by alalazo** [comment]
What is the status of this PR?

**Comment by alecbcs** [comment]
Going to go ahead and close this PR due to a lack of response. @francinelapid happy to reopen if you're still working on this PR.

**Pull request closed without merging**